src/rule_definition_tools.jl

```diff
@@ -357,7 +359,7 @@ function tuple_expression(primal_sig_parts)
     else
         num_primal_inputs = length(primal_sig_parts) - 1 # - vararg
         length_expr = :($num_primal_inputs + length($(esc(_unconstrain(primal_sig_parts[end])))))
-        :(ntuple(::Any -> DoesNotExist(), $length_expr))
+        @strip_linenos :(ntuple(::Any -> DoesNotExist(), $length_expr))
```

Comment:
I missed that `::Any -> DoesNotExist()` reintroduced a line node, in https://github.com/JuliaDiff/ChainRulesCore.jl/pull/325 I only inspected the output after `Base.remove_linenums!` was applied. The generated code now looks similar to
```julia
        function (::(ChainRulesCore.Core).kwftype(ChainRulesCore.typeof(ChainRulesCore.rrule)))(var"##kwargs#312"::ChainRulesCore.Any, ::ChainRulesCore.typeof(ChainRulesCore.rrule), ::(Core).Typeof(length), x::AbstractVector...)
            return (length(x...; var"##kwargs#312"...), begin
                        function length_pullback($(Expr(:meta, :nospecialize, :_)))
                            return ChainRulesCore.ntuple(((::ChainRulesCore.Any,)->begin
                                            ChainRulesCore.DoesNotExist()
                                        end), 1 + ChainRulesCore.length(x))
                        end
                    end)
        end
        function (ChainRulesCore.ChainRulesCore).rrule(::(Core).Typeof(length), x::AbstractVector...)
            #= REPL[17]:1 =#
            return (length(x...), begin
                        function length_pullback($(Expr(:meta, :nospecialize, :_)))
                            return ChainRulesCore.ntuple(((::ChainRulesCore.Any,)->begin
                                            ChainRulesCore.DoesNotExist()
                                        end), 1 + ChainRulesCore.length(x))
                        end
                    end)
        end
```

Reply:
BTW I just noticed that in the code for `Core.kwftype(...)` no line number is added. This was not introduced by https://github.com/JuliaDiff/ChainRulesCore.jl/pull/325 but existed already on the master branch - maybe it should be changed though?